src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -188,31 +191,34 @@ async def do_update(self, job, data):
 
         return await self.config()
 
-    @accepts(Bool('mount', default=True), Str('exclude_pool', default=None, null=True))
+    @accepts(
+        Bool('mount', default=True),
+        Str('exclude_pool', default=None, null=True),
+        Bool('boot_pool', default=None, null=True),
+        Bool('on_active', default=False),
```

Comment:
What is the purpose of this flag? The behavior was as if it was true by default, by here I see `default=False`

Reply:
For example, on a fresh HA system that has no zpools, the first zpool that gets created this logic will be executed. There is no reason to check `failover.status` in that scenario.